standalone/pruntime/enclave/src/contracts/btc_lottery.rs

```diff
@@ -0,0 +1,359 @@
+use crate::contracts;
+use crate::contracts::AccountIdWrapper;
+use bitcoin;
+use bitcoin::blockdata::script::Builder;
+use bitcoin::blockdata::transaction::{OutPoint, SigHashType, TxIn, TxOut};
+use bitcoin::consensus::encode::serialize;
+use bitcoin::network::constants::Network;
+use bitcoin::secp256k1::{All, Message, Secp256k1, Signature};
+use bitcoin::util::bip32::ExtendedPrivKey;
+use bitcoin::{Address, PrivateKey, PublicKey, Script, Transaction, Txid};
+use core::str::FromStr;
+use lazy_static;
+use parity_scale_codec::{Decode, Encode};
+use serde::{Deserialize, Serialize};
+use sp_core::crypto::Pair;
+use sp_core::ecdsa;
+extern crate runtime as chain;
+use crate::std::collections::BTreeMap;
+use crate::std::string::String;
+use crate::std::vec::Vec;
+use crate::types::TxRef;
+use crate::TransactionStatus;
+use rand::Rng;
+use sgx_rand::sample;
+use sp_core::hashing::blake2_128;
+use sp_core::U256;
+
+type SequenceType = u64;
+const ALICE: &'static str = "d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d";
+const RBF: u32 = 0xffffffff - 2;
+const LOTTERY: &'static str = "BTC LOTTERY";
+lazy_static! {
+    // 10000...000, used to tell if this is a NFT
+    static ref TYPE_NF_BIT: U256 = U256::from(1) << 255;
+    // 1111...11000...00, used to store the type in the upper 128 bits
+    static ref TYPE_MASK: U256 = U256::from(!u128::MIN) << 128;
+    // 00...00011....11, used to get NFT index in the lower 128 bits
+    static ref NF_INDEX_MASK: U256 = U256::from(!u128::MIN);
+}
+
+#[derive(Serialize, Deserialize, Default, Clone)]
+pub struct BtcLottery {
+    round_id: u32,
+    token_set: BTreeMap<u32, Vec<String>>,
+    lottery_set: BTreeMap<u32, BTreeMap<String, PrivateKey>>,
+    sequence: SequenceType,
+    queue: Vec<BtcTransferData>,
+    #[serde(skip)]
+    secret: Option<ecdsa::Pair>,
+    // Txid, vout, amount
+    utxo: BTreeMap<u32, BTreeMap<Address, (Txid, u32, u64)>>,
+    admin: Vec<AccountIdWrapper>,
+}
+
+impl core::fmt::Debug for BtcLottery {
+    fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
+        write!(f, "Hi")
+    }
+}
+
+/// These two structs below are used for transferring messages to chain.
```

Comment:
"///" is the docstring for an item (struct, function, etc). Use "//" for ordinary comments.